lib/audit-logs/src/lib.rs

```diff
@@ -28,5 +28,206 @@
 pub mod pg;
 mod stream;
 
+use pg::AuditDatabaseContext;
+use serde::Deserialize;
+use serde::Serialize;
+use si_data_pg::PgError;
+use si_data_pg::PgPoolError;
+use si_events::Actor;
+use si_events::ChangeSetId;
+use si_events::ChangeSetStatus;
+use si_events::UserPk;
+use si_events::WorkspacePk;
+use strum::Display;
+use strum::EnumDiscriminants;
+use telemetry::prelude::*;
+use thiserror::Error;
+
 pub use stream::AuditLogsStream;
 pub use stream::AuditLogsStreamError;
+
+#[allow(missing_docs)]
+#[derive(Debug, Error)]
+pub enum AuditLogError {
+    #[error("pg error: {0}")]
+    Pg(#[from] PgError),
+    #[error("pg pool error: {0}")]
+    PgPool(#[from] PgPoolError),
+    #[error("serde json error: {0}")]
+    SerdeJson(#[from] serde_json::Error),
+}
+
+type Result<T> = std::result::Result<T, AuditLogError>;
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AuditLog {
+    pub actor: Actor,
+    pub kind: AuditLogKind,
+    pub entity_name: String,
+    pub timestamp: String,
+    pub change_set_id: Option<ChangeSetId>,
+}
```

Comment:
This mirrors the type in `si-frontend-types` and `si-events`, but will be ultimately replaced.